### PR TITLE
Update `to_csv` docstring

### PR DESCRIPTION
**Pull request by scharlottej13**
- [x] partially addresses #9088 
- [ ] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by scharlottej13** [comment]
I chose not to list out the binary file modes (i.e. `wb` or `ab`), because I thought that could lead to more confusion and would likely break (https://docs.python.org/3/library/functions.html#open). Also wondering if we should reference [fsspec open_files](https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.open_files) too?

**Comment by MrPowers** [comment]
@scharlottej13 - this looks like a nice improvement to the docs 👍

**Review by pavithraes — APPROVED** [review]

**Comment by pavithraes** [comment]
@scharlottej13 Thanks, this looks good to me!

> Also wondering if we should reference [fsspec open_files](https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.open_files) too?

I think we can, maybe under the "See Also" heading, alongside `pandas` `to_csv`? -- What you have here works as-is too. :)

**Comment by scharlottej13** [comment]
thx @pavithraes @MrPowers! just added in the reference fsspec

File: docs/source/conf.py [hunk 1]
```diff
@@ -335,6 +335,10 @@
         "https://zarr.readthedocs.io/en/latest/objects.inv",
     ),
     "skimage": ("https://scikit-image.org/docs/dev/", None),
+    "fsspec": (
+        "https://filesystem-spec.readthedocs.io/en/latest/",
+        "https://filesystem-spec.readthedocs.io/en/latest/objects.inv",
+    ),
```
Comment (jrbourbeau) [inline docs/source/conf.py:341]: Nice -- thanks for adding this

**Review by jrbourbeau — COMMENTED** [review]
Thanks @scharlottej13!

File: dask/dataframe/io/csv.py [hunk 2]
```diff
@@ -848,8 +849,13 @@ def to_csv(
...
         defaults to 'ascii' on Python 2 and 'utf-8' on Python 3.
-    mode : str
-        Python write mode, default 'w'
+    mode : str, default 'w'
+        Python file mode. The default is 'w' (or 'wt'), for writing
+        a new file or overwriting an existing file in text mode. 'a'
+        (or 'at') will append to an existing file in text mode or
+        create a new file if it does not already exist.
+        Note that not all filesystems support the
+        append mode, especially on cloud storage systems.
```
[the remark this answers is not in the document]
Comment (scharlottej13) [reply dask/dataframe/io/csv.py:857]: Re appending-- I had the same question! Turns out cloud storage objects are "immutable objects" which is why they don't support append operations (more info in [GCP docs](https://cloud.google.com/storage/docs/key-terms#immutability)) *but* after I tested this, it turns out fsspec already has a [workaround](https://github.com/fsspec/s3fs/blob/fa1c76a3b75c6d0330ed03c46c2a7c848ebdb62e/s3fs/core.py#L1774-L1786)! so tl;dr, this works and maybe is something fsspec should brag about?

To your other question-- I think linking to Python's built-in `open` is great. I hadn't noticed this done in other docstrings, so I thought maybe there was some reason to not do this.

**Comment by jsignell** [comment]
I'm going to go ahead and merge this since it's clearly an improvement :) - thanks @scharlottej13!

**Pull request merged**